File: python/2315-count-asterisks.py
class Solution:
    def countAsterisks(self, s: str) -> int:
        bar_split = s.split("|")
        count = 0
    
        for i in range(0, len(bar_split), 2):
            count += bar_split[i].count("*")

        return count


s = '|**|*|*|**||***||'
print(Solution().countAsterisks(s))

File: javascript/2315-count-asterisks.js
/**
 * @param {string} s
 * @return {number}
 */
const countAsterisks = function(s) {
    const barSplit = s.split('|')
    let count = 0

    for (let i = 0; i < barSplit.length; i++) {
        if (i % 2 === 0) {
            for (const char of barSplit[i]) {
                if (char === '*') count++
            }
        }
    }
    return count
}

// const s = 'l|*e*et|c**o|*de|' // 2
// const s = 'yo|uar|e**|b|e***au|tifu|l' // 5
const s = '|**|*|*|**||***||' // 6
console.log(countAsterisks(s))